pennylane/qchem/factorization.py

```diff
@@ -14,31 +14,80 @@
 """
 This module contains the functions needed for two-electron tensor factorization.
 """
+from functools import partial
+
 import numpy as np
+import scipy as sp
 
 import pennylane as qml
 
-
-def factorize(two_electron, tol_factor=1.0e-5, tol_eigval=1.0e-5):
+has_jax_optax = True
```

Comment:
Yes, we are using `optax` as a soft dependency to use this functionality. Reasons I have to not just support `scipy`:
1. Efficiency - Automatic differentiation workflow turned out to be more efficient, especially when used with `jit` complied loss computation. This might not be directly seen for smaller molecules but as you go to medium ones, you begin noticing the benefits. 
2. Flexibility - Controlling the convergence with `scipy` solvers seems more difficult than the `optax` ones, especially when convergence is not reached with the default settings.
3. Maintainability - Using `scipy` would require implementing gradient computation for different options (for example type of regularization), which is done in a more straightforward way and is easier to extend.

Reply:
In an ideal world, we would need to add a soft dependency. But I think it's our best option in this case :+1: